src/sst.rs

```diff
@@ -253,6 +253,19 @@ impl SsTableFormat {
             self.compression_codec,
         )
     }
+
+    pub(crate) fn estimate_sst_size(
+        &self,
+        num_entries: usize,
+        total_entry_size_bytes: usize,
+    ) -> usize {
+        // u16 for key size, u32 for value size
```

Comment:
I think this logic should include the size for block metadata, the SST index, and the filter. See the `EncodedSsTableBuilder.build` method for the logic that creates these.

I also think we should exclude compression in this calculation. You haven't included it, so you're good here. I just want to make a note. Compression ratios are going to be hard to guess unless we track previous SSTs. I'd rather just define that the semantics of this function estimate the _uncompressed_ size, as you have done.